### PR TITLE
mediatek: add support for Acer Predator W6d and Acer Vero W6m

### PR DESCRIPTION
**Pull request by goldwrt**
This patch adds support for two variants of the already supported router
Acer Predator Connect W6: The Acer Predator Connect W6d (W6 without 6 GHz
wifi) and the Acer Connect Vero W6m (W6 without 2.5G eth1 port, usb3 port,
and the 6 on-board gpio RGB LEDs, and with a KTD2026 RGB LED controller
instead of the KTD2061 LED controller of the W6/W6d). Therefore, all
device tree information common to W6, W6d, and W6m is contained in file
`acer-w6-common.dtsi` which is included in the three device specific dts
files.

All suggestions of PR #16410 to improve current W6 support are adopted:
- reading device mac addresses from u-boot environment
- avoiding the detour via the file system to set the mac addresses
- correcting gpio LED labels
- shorter dummy `dm-mod.create` string in bootargs
- enable i2c interface

In addition, the dummy `dm-mod.create` string and the product name “Acer
Predator Connect W6” are corrected, and the LED configuration is changed
to reflect RX/TX activity and boot status.

The device tree for the W6m refers to the KTD202x driver suggested in
PR #16860.

A diff of the current `fdt-1` dtb for the W6 and the `fdt-1` created by
this patch for the W6 proves that they are equivalent except for the
changes described above and some line orders.

This patch has been extensively tested with some W6 and some W6m routers
since snapshot r27993 (up to r28000) without any boot fails or any crashes
occurring.

Signed-off-by: George Oldfort <openwrt@10099.de>

### CODE REVIEW

**Comment by jarledo** [comment]
Hi @goldwrt, thanks for taking the time and combining these common devices into a common .dtsi file. Have you looked at the compiled .dtb files per device to verify if they're combined correctly? I have two w6d devices I can help use for testing, both have been running great using PR #16410 rebased on top of main for a few weeks.

I see you also enabled I2C interface, have you used it for anything yet? I would be very interested in finding a way to disable the crazy bright "mood LED".

File: target/linux/mediatek/image/filogic.mk [hunk 1]
```diff
@@ -152,11 +152,11 @@ TARGET_DEVICES += acelink_ew-7886cax
 
 define Device/acer_predator-w6
   DEVICE_VENDOR := Acer
-  DEVICE_MODEL := Predator W6
+  DEVICE_MODEL := Predator Connect W6
   DEVICE_DTS := mt7986a-acer-predator-w6
   DEVICE_DTS_DIR := ../dts
   DEVICE_DTS_LOADADDR := 0x47000000
-  DEVICE_PACKAGES := kmod-usb3 kmod-mt7915e kmod-mt7916-firmware kmod-mt7986-firmware mt7986-wo-firmware e2fsprogs f2fsck mkf2fs
+  DEVICE_PACKAGES := kmod-usb3 kmod-i2c-gpio kmod-mt7915e kmod-mt7916-firmware kmod-mt7986-firmware mt7986-wo-firmware e2fsprogs f2fsck mkf2fs i2c-tools
```
Comment (musashino205) [inline target/linux/mediatek/image/filogic.mk:159]: Why do we need `kmod-i2c-gpio` for `DEVICE_PACKAGES`? The all dts files are not using `i2c-gpio`.

And, `i2c-tools` shouldn't be added if it's not used for the basic usage of the device.

File: target/linux/mediatek/dts/mt7986a-acer-predator-w6d.dts [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+		led-failsafe = &led_status_blue;
+		led-running = &led_status_green;
+		led-upgrade = &led_status_red;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		led_status_red: led0 {
+			label = "ant0:red";
```
Comment (musashino205) [inline target/linux/mediatek/dts/mt7986a-acer-predator-w6d.dts:22]: Please convert `label` properties of LEDs to `color` and `function`.

```
color = <LED_COLOR_ID_*>;
function = LED_FUNCTION_*;
function-enumerator = <N>;
```

or 

```
color = <LED_COLOR_ID_*>;
function = "function";
function-enumerator = <N>;
```

And, node names of LEDs should be `led-<N>` format.

https://elixir.bootlin.com/linux/v6.6.59/source/Documentation/devicetree/bindings/leds/common.yaml

BTW, `"<color>:<func>"` format should be used for `label` property.

**Comment by musashino205** [comment]
Please remove the merge commits.

**Comment by musashino205** [comment]
Please remove the merge commits instead of adding revert commits.

**Comment by jarledo** [comment]
I rebased this commit on top of `main` (commit 85844cfc5c591d00997ade665340315d54bc5822) and flashed one of my w6d devices today. Everything seems to work well, and thanks for your "sledgehammer" script to disable the mood leds! I don't have a serial device connected now, so I don't have a serial boot log to attach. Anything else I could do to help with testing?

**Comment by hauke** [comment]
Could you split this into multiple commits to make the review easier please.

1. commit extraction into target/linux/mediatek/dts/mt7986a-acer-w6-common.dtsi without big changes
2. updates to the device tree
3. add new devices

**Comment by jarledo** [comment]
Hi @goldwrt, I tested your latest split commits. This time I installed it on both my w6d devices, one with a 2022.X uboot version and the other one with a 2023.X. I didn't spot any errors in the system logs so far, and everything seems to be working well. Thanks for your contribution!

**Pull request merged**